src/Libraries/RevitNodesUI/Selection.cs

```diff
@@ -49,16 +49,22 @@ public RevitDynamoModel RevitDynamoModel
             {
                 if (revitDynamoModel != null)
                 {
-                    var hwm = revitDynamoModel.Workspaces.OfType<HomeWorkspaceModel>().ElementAt(0);
-                    hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
+                    var hwm = revitDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+                    if (hwm != null)
+                    {
+                        hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
```

Comment:
@aparajit-pratap 
Thanks for the comment!

If registrition or unregistrition happens in the workspace, that means every time when a file is opened, we need to check whether there are selection nodes, if so, register the event. This is not necessary for files which do not contain selection nodes. Also if selection nodes are added/removed (users are free to edit a file by adding or removing nodes), the event handler need to be registered/unregistered. At this time, the workspace is open and it will not be removed when adding/removing nodes happens.